### DISCUSSION TITLE
Is there any way to add a non-existing network to the wallet?

### DISCUSSION THREAD

**Discussion by ghost** [wevm/wagmi#1670] [Q&A]
I can't find such a function in wagmi.

**Comment by tmm** [comment]
[`useSwitchNetwork`](https://wagmi.sh/react/hooks/useSwitchNetwork) allows you to switch to a different network and adds the network if not present.

**Reply by ghost** [reply]
I understand that there is no such function in vanilla js?

**Reply by tmm** [reply]
There is: [`switchNetwork`](https://wagmi.sh/core/actions/switchNetwork). Please take a look at the docs.

**Reply by ghost** [reply]
I've seen this function, but it doesn't add a non-existing network, it just switches it.

**Reply by tmm** [reply]
It depends on the `Connector` you are using and whether it supports the `wallet_addEthereumChain` RPC method.
